### PR TITLE
new syntax: Interface NAME from JSON_FILE

### PR DESCRIPTION
**Pull request by christianparpart**
Addressing https://github.com/ethereum/solidity/issues/1687

This is work-in-progress.

### CODE REVIEW

File: docs/Solidity.g4 [hunk 1]
```diff
@@ -36,10 +46,25 @@ importDirective
 importDeclaration
   : identifier ('as' identifier)? ;
 
+interfaceDefinition
+  : 'interface' identifier 'from' StringLiteralFragment
+  | 'interface' identifier 'as' AbiString
```
Comment (chriseth) [inline docs/Solidity.g4]: I don't think we should support interfaces from abi strings.

File: libsolidity/ast/AST.h [hunk 2]
```diff
@@ -172,6 +173,9 @@ class SourceUnit: public ASTNode
 	std::optional<std::string> const& licenseString() const { return m_licenseString; }
 	std::vector<ASTPointer<ASTNode>> nodes() const { return m_nodes; }
 
+	/// Replaces given top-level AST node with a new AST node.
+	void replaceNode(ASTPointer<ASTNode> _oldNode, ASTPointer<ASTNode> _newNode);
```
Comment (chriseth) [inline libsolidity/ast/AST.h]: I don't think we should modify the ast after the fact.

File: libsolidity/ast/JsonInterfaceImporter.cpp [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	// "indexed" (only when event)
+	auto const parameterName = _node["name"].asString();
+	auto const typeName = _node["type"].asString();
+	ASTPointer<TypeName> type = {}; // TODO (get me from typeName)
+
+	return createASTNode<VariableDeclaration>(
+		_location,
+		type,
+		make_shared<ASTString>(parameterName),
+		ASTPointer<Expression>{},			   // value
```
Comment (stackenbotten) [inline libsolidity/ast/JsonInterfaceImporter.cpp:140]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `600297038477404ed292b36d38d05eb3bd1208b9`:
```
Coding style error:
 libsolidity/ast/JsonInterfaceImporter.cpp:140: ASTPointer<Expression>{}, // value

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `66fa2f6400ec36cf51e05f574f46fb2afb2ff7ee`:
```
Coding style error:
 libsolidity/ast/JsonInterfaceImporter.cpp:140: ASTPointer<Expression>{}, // value

```
Please check that your changes are working as intended.

File: libsolidity/ast/JsonInterfaceImporter.cpp (hunk 3 above)
Comment (stackenbotten) [inline libsolidity/ast/JsonInterfaceImporter.cpp:140]: Coding style error

**Comment by chriseth** [comment]
Shall we close this?

**Pull request closed without merging**